### ISSUE TITLE
Correlation ID not sent to upstream backend

### ISSUE THREAD

**Issue by fiatfilip** [Haufe-Lexware/wicked.haufe.io#156]
Athough the auth server json file contains the the correlation id plugin setup
...{
        "name": "correlation-id",
        "config": {
          "header_name": "Correlation-Id",
          "generator": "uuid",
          "echo_downstream": false
        }
 }, ...
 the correlation-id header misses from the requests to backends.

**Comment by DonMartin76** [comment]
This refers to requests to the endpoints which the auth server uses to find scopes and similar things, correct?

**Comment by fiatfilip** [comment]
No, it refers to the "normal" backends of the enpoints served through the API gateway.
In the backend, the request contains no correlation-id header.

**Comment by DonMartin76** [comment]
But then you should specify the `Correlation-Id` plugin on those APIs, not on the Auth Server? The Auth Server is also served via the API Gateway (Kong), which is why you have the opportunity to alter those settings via the auth server config JSON, and what I wrote in my second comment is actually also true.

But the better, you should be fine by just adding the same plugin to your actual APIs.

**Comment by DonMartin76** [comment]
The issue that the auth server does not pass on correlation IDs to the scope decision/external user verification service is still a bug, though.
